### PR TITLE
Provide copy option in extended message section of Confirm dialog

### PR DESCRIPTION
**Pull request by radhikav1**
# Provide copy option in extended message section of new Confirm dialog

## Description
Provided message part with copy option in extended message section of new Confirm dialog

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [x] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-20796](https://cdap.atlassian.net/browse/CDAP-20796)

## Test Plan
n/a

## Screenshots
image

[CDAP-2079]: https://cdap.atlassian.net/browse/CDAP-2079?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CDAP-20796]: https://cdap.atlassian.net/browse/CDAP-20796?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/TransformAdd.tsx
Comment (radhikav1) [inline app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/TransformAdd.tsx:1]: Indentation corrected. this file is not related to confirm dialog changes.

File: app/cdap/components/shared/ConfirmDialog/index.tsx [hunk 1]
```diff
@@ -68,12 +86,25 @@ export const ConfirmDialog = ({
...
 
+  const showCopyOption = () => {
+    copyToClipBoard(copyableExtendedMessage).then(
+      () => {
+        /* tslint:disable:no-console */
+        console.log('Success!. Text is copied to clipboard');
+      },
+      () => {
+        /* tslint:disable:no-console */
+        console.error('Fail!. Text copy is failed');
```
Comment (njbriggs) [inline app/cdap/components/shared/ConfirmDialog/index.tsx]: Can you remove the `console.log`s?

**Review by sumengwang — COMMENTED** [review]
Your jira link is missing a number

File: app/cdap/components/shared/ConfirmDialog/index.tsx [hunk 2]
```diff
@@ -23,8 +23,20 @@ import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
...
-import { StyledBox, StyledDialog, StyledAlert } from './styles';
+import { StyledBox, StyledDialog, StyledAlert, CopyContentBox } from './styles';
 import PrimaryTextButton from 'components/shared/Buttons/PrimaryTextButton';
+import { Color } from '@material-ui/lab/Alert';
+import { copyToClipBoard } from 'services/Clipboard';
+import FileCopyOutlinedIcon from '@material-ui/icons/FileCopyOutlined';
+
+export type Severity = Color;
+
+export enum SeverityType {
```
Comment (sumengwang) [inline app/cdap/components/shared/ConfirmDialog/index.tsx:31]: where is this used?

File: app/cdap/components/shared/ConfirmDialog/index.tsx [hunk 3]
```diff
@@ -68,12 +86,25 @@ export const ConfirmDialog = ({
     }
   };
 
+  const showCopyOption = () => {
```
Comment (sumengwang) [inline app/cdap/components/shared/ConfirmDialog/index.tsx]: why is it named `showCopyOption`? Isn't is doing copy to clipboard?

File: app/cdap/components/shared/ConfirmDialog/index.tsx (hunk 2 above)
Comment (radhikav1) [reply app/cdap/components/shared/ConfirmDialog/index.tsx:31]: This would be used by consumer components of this Confirm dialog. Something like below, 
```
<ConfirmDialog
      headerTitle={'Delete'}
     ....
      cancelButtonText={T.translate(`${PREFIX}.cancelButtonText`)}
      confirmButtonText={T.translate(`${PREFIX}.deleteButtonText`)}
     .....
      severity={SeverityType.ERROR}
      statusMessage={deleteErrorMsg}  >
</ConfirmDialog>
```
As this is used for Confirm Dialog , enum is defined inside the same. Pls let me know if there are any suggestions here.

File: app/cdap/components/shared/ConfirmDialog/index.tsx (hunk 1 above)
Comment (radhikav1) [reply app/cdap/components/shared/ConfirmDialog/index.tsx]: Removed

File: app/cdap/components/shared/ConfirmDialog/index.tsx (hunk 3 above)
Comment (radhikav1) [reply app/cdap/components/shared/ConfirmDialog/index.tsx]: Changed the handling way now.

**Comment by radhikav1** [comment]
> Your jira link is missing a number

Corrected, Thanks for pointing out. :)

File: app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/TransformAdd.tsx [hunk 4]
```diff
@@ -135,10 +135,12 @@ export default function TransformAddButton({
           dense: true,
         }}
       >
-        <MenuItem onClick={(event) => {
-          handleMenuClick('rename');
-          setSubMenuAnchorEl(event.currentTarget);
-        }}>
+        <MenuItem
+          onClick={(event) => {
```
Comment (njbriggs) [inline app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/TransformAdd.tsx:139]: Can you remove these changes? I don't think they're relevant to this PR.

File: app/cdap/components/shared/ConfirmDialog/index.tsx (hunk 2 above)
Comment (njbriggs) [reply app/cdap/components/shared/ConfirmDialog/index.tsx:31]: But the `severity` field takes type `Severity` not `SeverityType`?

File: app/cdap/components/shared/ConfirmDialog/styles.ts [hunk 5]
```diff
@@ -48,3 +53,10 @@ export const StyledBox = styled(Box)`
     padding: 0px 8px;
   }
 `;
+
+export const CopyContentBox = styled(Box)`
+  background-color: #0000001a;
```
Comment (GnsP) [inline app/cdap/components/shared/ConfirmDialog/styles.ts]: we should avoid using string literals like `#0000001a`.

**Review by GnsP — APPROVED** [review]

File: app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/TransformAdd.tsx (hunk 4 above)
Comment (radhikav1) [reply app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/TransformAdd.tsx:139]: you are right , these are not related to current PR. But indentation being corrected automatically and got added to PR.

File: app/cdap/components/shared/ConfirmDialog/index.tsx (hunk 2 above)
Comment (radhikav1) [reply app/cdap/components/shared/ConfirmDialog/index.tsx:31]: Streamlined the code now, Using SeverityType instead of Severity.

File: app/cdap/components/shared/ConfirmDialog/styles.ts (hunk 5 above)
Comment (radhikav1) [reply app/cdap/components/shared/ConfirmDialog/styles.ts]: changed

**Review by sumengwang — APPROVED** [review]
